src/com/pentalog/pentastagiu/homework5/HomeworkApp.java

```diff
@@ -0,0 +1,115 @@
+package com.pentalog.pentastagiu.homework5;
+
+import java.util.*;
+import java.util.logging.LoggingPermission;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class HomeworkApp {
+    public static void main(String[] args) {
+        System.out.println("Circle app");
+        circleListApp();
+        System.out.println("String set app");
+        stringSetApp();
+        System.out.println("Employees map app");
+        employeesMapApp();
+        System.out.println("Functional interface app");
+        NumberInterface firstTry = (Number nr1, Number nr2) -> {
+            return nr1.intValue()*nr2.intValue()+nr1.intValue()/nr2.intValue()-nr2.intValue();
+        };
+        NumberInterface secondTry = (Number nr1, Number nr2) -> {
+            return nr2.intValue()*nr1.intValue()+nr2.intValue()/nr1.intValue()-nr1.intValue();
+        };
+        NumberInterface thirdTry = (Number nr1, Number nr2) -> {
+            return nr1.intValue()*nr1.intValue()+nr1.intValue()/nr1.intValue()-nr2.intValue();
+        };
+        System.out.print("1: ");
+        System.out.println(operate(2, 1, firstTry));
+        System.out.print("2: ");
+        System.out.println(operate(1, 2, secondTry));
+        System.out.print("3: ");
+        System.out.println(operate(5, 4, thirdTry));
+    }
+
+    private static void circleListApp() {
+        List<Circle> circleList = new ArrayList<>();
+        double randomRadius=1;
+        while(randomRadius>0.01) {
+            randomRadius = Math.random();
+            circleList.add(new Circle(randomRadius));
+        }
+        for(Circle circle : circleList) {
+            System.out.println(circle.getArea());
+        }
+    }
+
+    private static void stringSetApp(){
+        Set<String> stringSet = new HashSet<>();
+        stringSet.add("Ana");
+        stringSet.add("are");
+        stringSet.add("mere");
+        stringSet.add("Andrei");
+        stringSet.add("nu are");
+        stringSet.add("pere");
+
+        System.out.println("Iterator");
+        Iterator iterator = stringSet.iterator();
+        while(iterator.hasNext()){
+            System.out.println(iterator.next());
+        }
+
+        System.out.println("For iteration");
+        for(String string : stringSet) {
+            System.out.println(string);
+        }
+
+        System.out.println("Stream");
+        String strings = stringSet
+                .stream()
+                .map(elem -> elem.toLowerCase())
```

Comment:
Filter are posibilitatea de a reduce dimensiunea strem-ului astfel reducand numarul de operatii executate. De aceea ar trebui pus inainte de map pentru a eficientiza algoritmul. Astfel elementele care nu trec de filter nu mai ajung in map